src/util.h

```diff
@@ -82,28 +82,34 @@
 
 #else
 
+void* (*git__malloc_func)(size_t size) = malloc;
+void* (*git__calloc_func)(size_t nitems, size_t size) = calloc;
+char* (*git__strdup_func)(const char *str1) = strdup;
+void* (*git__realloc_func)(void *ptr, size_t size) = realloc;
+void* (*git__free_func)(void *ptr) = free;
+
```

Comment:
All of these assignments need to be in a .c file. The header can then have an extern definition